src/proto/grpc/testing/control.proto

```diff
@@ -92,28 +73,82 @@ message SecurityParams {
 message ClientConfig {
   // List of targets to connect to. At least one target needs to be specified.
   repeated string server_targets = 1;
-  ClientType client_type = 2;
+
   SecurityParams security_params = 3;
-  // How many concurrent RPCs to start for each channel.
-  // For synchronous client, use a separate thread for each outstanding RPC.
-  int32 outstanding_rpcs_per_channel = 4;
+  // Maximum limit of how many concurrent RPCs in flight at any time
+  // for each channel.
+  // For synchronous clients, typically use a separate thread of control
+  // for each outstanding RPC.
+  // In practice, this parameter can limit QPS for closed-loop and open-loop
+  // tests. This value should be set to at least the Little's law value for your
+  // desired QPS: namely at least QPS/channel * message latency .
+  int32 max_outstanding_rpcs_per_channel = 4;
+
   // Number of independent client channels to create.
   // i-th channel will connect to server_target[i % server_targets.size()]
   int32 client_channels = 5;
-  // Only for async client. Number of threads to use to start/manage RPCs.
+
+  // Allow the driver to specify the number of threads to use to
+  // start/manage RPCs. This only applies in cases where the client thread
+  // count is not dependent on the RPC width or depth (e.g., the async client
+  // case in C++),
+  // The worker may ignore this if it is not relevant. The value 0 (or not
+  // present) should be used as an instruction to auto-size.
   int32 async_client_threads = 7;
+
+  // Specify the desired type of RPC to use
   RpcType rpc_type = 8;
-  // The requested load for the entire client (aggregated over all the threads).
+
+  // The requested load for this client worker.
   LoadParams load_params = 10;
+
+  // Information about what to send/expect in the payload
   PayloadConfig payload_config = 11;
+
+  // Parameters for setting up the statistics histogram response
   HistogramParams histogram_params = 12;
 
   // Specify the cores we should run the client on, if desired
   repeated int32 core_list = 13;
   int32 core_limit = 14;
+
+  // What the driver calls the scenario
+  string scenario_name = 15;
+
+  // Which version of the API for this language should the client use?
+  // e.g. in C++, the valid options might be "sync" and "async"
+  // In any language, the valid options are specified in the
+  // response to the Capabilities RPC
+  string client_api = 16;
+}
+
+message DriverCommandAccepted {
+  // The worker can give an optional explanation to the driver
+  string explanation = 1;
+}
+
+message DriverCommandUnimplemented {
+  // The worker can give an optional explanation to the driver
+  string explanation = 1;
+}
+
+message DriverResponse {
```

Comment:
This message seems to be an overkill. I think all we need is if  the worker doesn't support given scenario, it will reply with statusCode e.g.  INVALID_ARGUMENT  and send explanation as a status detail (and qps driver to properly handle such response by skipping the scenario).
Afterall, I don't expect us to be trying to run scenarios that are not supported.